common/src/lib.rs

```diff
@@ -99,3 +99,24 @@ impl BlockHeader {
         Ok(())
     }
 }
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct MerkleProof {
+    // TODO
+}
+
+#[derive(Error, Debug, Serialize, Deserialize, Clone)]
+pub enum MerkleProofError {
+    /// When the proof is malformed.
+    #[error("malformed proof: {0}")]
+    MalformedProof(String),
+    /// When the root doesn't match
+    #[error("unmatched string: found {0} but expected {1}")]
```

Comment:
How about `expected {1}, but found {0}`? I think I've seen more of the error messages like this.